dashboard/test/controllers/report_abuse_controller_test.rb

```diff
@@ -0,0 +1,183 @@
+require 'test_helper'
+require 'cdo/aws/s3'
+
+class ReportAbuseControllerTest < ActionController::TestCase
+  include Devise::Test::ControllerHelpers
+
+  setup do
+    # channels
+    # Use a fake storage ID, because if you use the project_storage factory,
+    # everything breaks.
+    storage_id = 1
+    @projects = Projects.new(storage_id)
+    @channel_id = @projects.create({}, ip: '10.0.0.1')
+    @controller.stubs(:get_storage_id).returns(@storage_id)
+
+    # files
+    AWS::S3.stubs :create_client # Don't actually talk to S3
+    FileBucket.any_instance.stubs(:list).returns([{filename: 'test.file'}])
+  end
+
+  teardown do
+    @controller.unstub(:get_storage_id)
+
+    FileBucket.any_instance.unstub(:list)
+    FileBucket.any_instance.unstub(:get_abuse_score)
+  end
```

Comment:
please add some tests covering the various authorization scenarios for the endpoints we are still exposing (show_abuse and reset_abuse), making sure they do the right thing for (signed out user, signed in user, project validator). we have a `test_user_gets_response_for` helper method that makes this into a one-liner. Here are some examples: https://github.com/code-dot-org/code-dot-org/blob/f0c7f6031075887eaf13f18de05c4cecc1a7c1db/dashboard/test/controllers/data_docs_controller_test.rb#L30-L34

there is already a `:project_validator` factory which you can use instead of `:levelbuilder` from the above examples.

Reply:
I've updated the names for the tests to be clearer, and added one for resetting as a student, but I think these cases were already mostly covered: https://github.com/code-dot-org/code-dot-org/pull/49960/files#diff-8750f4b622771024bbca5eea8a621d87d9aaedc5f40d19352ff153fb050b9294R96-R134. The `show_abuse` endpoint is available to all users, including signed-out. Am I missing anything here?